### PR TITLE
ARGO-105 Fix response headers on update/delete

### PR DESCRIPTION
**Pull request by kaggis**
- Issues Found: 
 - That in some cases in update/delete responses (Avail.Profiles) the response header wasn't correctly formatted (Fixed)
 - That when an id was provided for update/delete profile it was forwarded to the query before checking that the string is a valid ObjectID string. 
 - Added proper xml responses when a profile is properly deleted or updated (Were not implemented)
 - Added checks when poem definitions are missing in a profile to properly display results with an empty "poem" string

### CODE REVIEW

**Comment by kaggis** [comment]
please review
/cc @pkoro @skanct

**Comment by pkoro** [comment]
+1 from my side @kaggis . Once it is also tested on devel node we should merge in order to deploy to production.

**Comment by kaggis** [comment]
Adding Unit testing in availabilty profiles requests. 
 - Init testing enviroment (test database, seed values)
 - Test Create Profile

...other tests in the commits to follow

**Comment by kaggis** [comment]
Added response message when user submits malformed json data in POST,PUT

Added comprehensive unit tests:
- Create Av.Profiles
- Read Av.Profile List
- Update Av.Profiles
- Delete Av.Profiles
- Tests for using wrong profile ids
 - in Update request
 - in Delete request
- Tests for using bad json input
 - in Create request
 - in Update request
- Tests for bad authorization
 - in Create request
 - in Update request
 - in Delete request

Added comprehensive comments and go doc strings:


please review 
/cc @skanct @pkoro @dpavlos

File: app/availabilityProfiles/availabilityProfiles_test.go [hunk 1]
```diff
@@ -0,0 +1,555 @@
...
+	cfg                 config.Config
+	resp_profileCreated string
+	resp_profileUpdated string
+	resp_profileDeleted string
+	resp_unauthorized   string
+	resp_no_id          string
+	resp_bad_json       string
+}
+
+// Setup the Test Enviroment
```
Comment (dpavlos) [inline app/availabilityProfiles/availabilityProfiles_test.go]: Couple of minor typos eg "enviroment" "Also here is are instantiated"

File: app/availabilityProfiles/availabilityProfiles_test.go [hunk 2]
```diff
@@ -0,0 +1,555 @@
...
+	// Insert first seed profile
+	c.Insert(bson.M{"name": "ap2", "namespace": "namespace2", "poems": []string{"poem02"},
+		"groups": [][]string{
+			[]string{"ap2-service1", "ap2-service2", "ap2-service3"},
+			[]string{"ap2-service4", "ap2-service5", "ap2-service6"}}})
+
+}
+
+// Testing creation of a profile  using POST request.
+// During Setup of the test enviroment the testdb is seeded with
```
Comment (dpavlos) [inline app/availabilityProfiles/availabilityProfiles_test.go]: Typo "enviroment"

File: app/availabilityProfiles/availabilityProfiles_test.go [hunk 3]
```diff
@@ -0,0 +1,555 @@
...
+	defer session.Close()
+	// Open collection aps
+	c := session.DB(suite.cfg.MongoDB.Db).C("aps")
+	// Remove the specific profile inserted during this test
+	c.Remove(bson.M{"name": "fresh_test_profile"})
+
+}
+
+// Testing Reading of profile list using GET request.
+// During Setup of the test enviroment the testdb is seeded with
```
Comment (dpavlos) [inline app/availabilityProfiles/availabilityProfiles_test.go]: typo

File: app/availabilityProfiles/availabilityProfiles_test.go [hunk 4]
```diff
@@ -0,0 +1,555 @@
...
+        "groups": [
+            [
+               "service1",
+               "service2",
+               "service3"
+            ],
+            [
+                "service4",
+                "service5",
+                "service6"
```
Comment (pkoro) [inline app/availabilityProfiles/availabilityProfiles_test.go]: Indentation here is a bit off (3 spaces instead of 2).

**Comment by pkoro** [comment]
+1 from me. My only comment other than the minor ones on typos and indentation is that maybe we should add in the README a one line on how to generate docs.

File: app/availabilityProfiles/availabilityProfiles_test.go (hunk 4 above)
Comment (kaggis) [inline app/availabilityProfiles/availabilityProfiles_test.go]: Regarding identation: it belongs inside a string litteral in opened ` `

**Pull request merged**